plugins/modules/license_keys.py

```diff
@@ -0,0 +1,291 @@
+from ansible.module_utils.basic import AnsibleModule
+
+from ..module_utils.sap_launchpad_systems_runner import *
+from ..module_utils.sap_id_sso import sap_sso_login
+
+
+DOCUMENTATION = r'''
+---
+module: license_keys
+
+short_description: Creates systems and license keys on me.sap.com/licensekey
+
+version_added: 1.1.0
+
+options:
+  suser_id:
+    description:
+      - SAP S-User ID.
+    required: true
+    type: str
+  suser_password:
+    description:
+      - SAP S-User Password.
+    required: true
+    type: str
+  installation_nr:
+    description: 
+      - Number of the Installation for which the system should be created/updated
+    required: true
+    type: str
+  system:
+    description:
+      - The system to create/update
+    required: true
+    type: dict
+    suboptions:
+      nr:
+        description:
+          - The number of the system to update. If this attribute is not provided, a new system is created.
+        required: false
+        type: str
+      product:
+        description:
+          - The product description as found in the SAP portal, e.g. SAP S/4HANA
+        required: true
+        type: str
+      version:
+        description:
+          - The description of the product version, as found in the SAP portal, e.g. SAP S/4HANA 2022
+        required: true
+        type: str
+      data:
+        description: 
+          - The data attributes of the system. The possible attributes are defined by product and version.
+          - Running the module without any data attributes will return in the error message which attributes are supported/required.
+        required: true
+        type: dict
+
+  licenses:
+    description:
+      - List of licenses to create for the system.
+      - If the license does not exist, it is created.
+      - If it exists, it is updated.
+    required: true
+    type: list
+    elements: dict
+    suboptions:
+      type:
+        description:
+          - The license type description as found in the SAP portal, e.g. Maintenance Entitlement
+        required: true
+        type: str
+      data:
+        description: 
+          - The data attributes of the licenses. The possible attributes are defined by product and version.
+          - Running the module without any data attributes will return in the error message which attributes are supported/required
+          - In practice, most license types require at least a hardware key (hwkey) and expiry date (expdate)
+        required: true
+        type: dict
+      
+  delete_other_licenses:
+    description:
+      - Whether licenses other than the ones specified in the licenses attributes should be deleted.
+      - This is handy to clean up older licenses automatically.
+    type: bool
+    required: false
+    default: false
```

Comment:
to make it more idempotent shouldn't there a switch e.g. 'state: absent | present' and then delete the defined one instead of deleting everything else.

Reply:
good point. the `delete_other_licenses` is inspired by a requirement that most customers will have. Imagine that you have some CI/CD pipeline that sets or upgrades your systems. at the end of the pipeline, when the VM is set up, as a last step of your pipeline, you will need to create a license for this system, meaning upserting the hwkey in me.sap.com using this module.

you use this module to register the latest hw key. you don't care about all the older licenses you had for this system, in fact you want to clean them up. this is what this switch does for you - it deletes all the licenses of this system except the one that was provided.

if we follow your suggested approach and delete the defined ones, how would one collect all the older, unused licenses that should be deleted in the use case above? one would need another module that allows us to retrieve all license keys of the system and filter all which were not provided by the user in the current run - which is exactly what the logic behind the `delete_other_licenses` flag does for you :)

Reply:
sounds reasonable, but shouldn't there the possibility to delete the whole system

Reply:
@rainerleber I thought that's a good idea and started adding deletion via `state: absent`. 

however, I hit quite a roadblock. the deletion via the portal seems to talk with a different API (`https://me.sap.com/backend` as opposed to `https://launchpad.support.sap.com` which this module supports). If I use the http client from this module (which did a login with `sap_sso_login`) I only receive 401s. I have no idea how we could extend support of this module to me.sap.com, but one would probably have to reverse engineer the current portal login again ;)

I'd suggest to leave out the delete case and merge what we have, since I think automating the deletion is less important than the creation.

you can find the WIP in https://github.com/MatthiasWinzeler/community.sap_launchpad/commit/ff9bb6e1908a13b38633702f5d914b6d3cc741a5, if you want to give it a try too.

Reply:
Thanks for the try. It's not a blocker yet just a backlog item ;-)